### ISSUE TITLE
Feature request: provide tabbed or dual view for site preview configured content types.

### ISSUE THREAD

**Issue by laurencefass** [chapter-three/next-drupal#217]
Ive just started working with this module so apologies if I have missed some configuration details on how to do this.

By following the  [introductory configuration for content types](https://next-drupal.org/learn/preview-mode/configure-content-types) lose the Drupal core output of a node only presented with the Next JS preview.

Is the only way to restore the Drupal core view to remove the configuration at /admin/config/services/next/entity-types?

Would it be possible to render both the Next JS output and the core Drupal output and have them available as separate tabs (or some other display type).

Thanks

**Comment by laurencefass** [comment]
yes, exactly that. presently the views are mutually exclusive and i have to delete configuration to restore the drupal core display. would it be possible to have them both?

**Comment by laurencefass** [comment]
no its not a blocker thanks for asking. 

at this stage i am just testing capaibilities seeing how everything works and fits together.

**Comment by TheRuslan** [comment]
This option could be useful.

**Comment by fiasco** [comment]
I've opened #241 to share what we've done to date. I think we'd want to extend it to 

- Allow the HtmlRenderer intervention to be disabled (so Drupal could still render a frontend if it wanted to.
- Support non-Node entities.

**Comment by shadcn** [comment]
OK so some updates here.

I implemented a POC for the headless tab (uses a new route `entity.ENTITY_TYPE.headless_preview`) and ran some tests. It looks like while this works for the default entity view, it introduces some complexity when dealing with non-canonical routes (example revision routes or preview routes. See @Elendev solution at #132).

Example: `node/12/headless` renders the latest revision for `node 12` but there's no way to easily handle revisions here.

I'm thinking of other ways we could solve this and I'm leaning towards a block-based solution.

- The `Headless Preview` will be available as a block.
- Instead of taking over `HtmlRenderer`, you will be able to enable the preview block and use visibility filters to turn it on/off.
- The block can come with a site switcher toggle where you can select between the default Drupal view (Twig) and headless views (Next.js)
- The site switcher can be its own block as well for more flexibility.

![headless-preview](https://user-images.githubusercontent.com/124599/176123361-e4b476f4-3bf8-4d74-9654-bd076dc5a0f3.png)

What do you think?

**Comment by laurencefass** [comment]
As the site select is an optional block, what is the default display under the view tab? Something would need to be displayed if the block is not visible.

Would this be configurable by next_drupal, the block, or the node?

**Comment by shadcn** [comment]
We can show the Twig preview by default (if available) or make this configurable by entity type.

**Comment by laurencefass** [comment]
+1 for configurable options. 

Is the Twig Preview the Drupal default display?

**Comment by shadcn** [comment]
> Is the Twig Preview the Drupal default display?

Yes or whatever the main content block render. Example the default display on a node view route or a revision display on a revision route.

**Comment by shadcn** [comment]
I moved this to the `2.0.0` release for now but if we can figure out how to implement this in a non-breaking way, we could consider it in a dot release.

**Comment by fiasco** [comment]
Being a block, would it only work on an entity view page? This wouldn't work in scenarios where Drupal was a hybrid CMS unless you relied solely on the site selector and viewed the site directly (no iframe, new tab) through that?

**Comment by shadcn** [comment]
@fiasco You'll be able to place the block on any canonical route for any entity type.

When in hybrid mode, the block will default to the default html renderer so you will see the Twig template. You can then use the site selector to switch sites.

**Comment by jmcerda** [comment]
We have been running a hook-based version of this switcher in production on Drupal 11 with `next` 2.1.0. It avoids the separate-route and revision-handling complexity discussed above by switching the render on the routes that `HtmlRenderer` already supports.

The approach is:

- implement `hook_next_site_preview_alter()` and use `$context['original_build']` for the Drupal-rendered view;
- keep the iframe and Drupal render on the same canonical, latest-version, and revision route, selected by an ephemeral query argument;
- permission-gate the control and add the query-argument cache context;
- preserve the existing iframe behavior by default, so enabling the feature can be backward-compatible;
- leave Drupal's core unsaved-form preview alone (we use a link to the saved-draft frontend preview there, because replacing the core preview would hide unsaved edits).

The current `main` branch still supplies `original_build` through this alter hook, so the extension point remains sufficient. Our public reference implementation is [`wl_next_render_toggle`](https://github.com/Wilkes-Liberty/webcms/tree/master/web/modules/custom/wl_next_render_toggle); it also contains site-specific presentation and frame controls that I would leave out of a focused upstream change.

Would maintainers be open to a small PR against `main` that adds a configurable, disabled-by-default render switcher using this seam, with coverage for the existing canonical/latest-version/revision behavior on the supported Drupal versions?
